lib/ivis_opengl/piedraw.cpp

```diff
@@ -379,9 +360,11 @@ static void pie_DrawShadow(iIMDShape *shape, int flag, int flag_data, const glm:
 
 	// draw the shadow volume
 	const auto &program = pie_ActivateShader(SHADER_GENERIC_COLOR, pie_PerspectiveGet() * modelViewMatrix, glm::vec4());
-	static glBufferWrapper buffer;
-	glBindBuffer(GL_ARRAY_BUFFER, buffer.id);
-	glBufferData(GL_ARRAY_BUFFER, sizeof(Vector3f) * vertexes.size(), vertexes.data(), GL_STREAM_DRAW);
+	static gfx_api::buffer* buffer;
+	if (buffer)
+		delete buffer;
+	buffer = gfx_api::context::get().create_buffer(gfx_api::buffer::usage::vertex_buffer, sizeof(Vector3f) * vertexes.size(), gfx_api::context::buffer_storage_hint::stream_draw);
```

Comment:
This is not equivalent code to the original code, and will introduce a performance regression. You are creating and deleting the buffer every time through. There is overhead to that.